### PR TITLE
Create Patients Table; Migrate Data

### PR DESCRIPTION
**Pull request by iamstevedavis**
Pull patients out into their own table. I figured we might end up needing to do this at some point anyway and it helped me get familiar with the app.

### CODE REVIEW

**Comment by iamstevedavis** [comment]
This will break https://github.com/KrisBraun/sanctuary-patient-api/pull/5 so merge that in first and I'll update the db call in this PR

File: migrations/20200327213301_create_patients_table.js [hunk 1]
```diff
@@ -0,0 +1,68 @@
+
+exports.up = function (knex) {
+  // Create new patient table
+  return knex.schema.createTable("patients", (table) => {
+    table.increments("id").primary();
+    table.text("name");
+    table.text("phone_number");
```
Comment (gingerich) [inline migrations/20200327213301_create_patients_table.js]: We should be consistent with casing. Looks like the current convention is camelCase

Comment (gingerich) [reply migrations/20200327213301_create_patients_table.js]: Also patient_id

Comment (iamstevedavis) [reply migrations/20200327213301_create_patients_table.js]: :+1:

**Comment by clark-teeple** [comment]
Would it make sense do an automatic upsert when an appointment is created? That way we don't add an extra step to the workflow for the people at Sanctuary, but we still get a record of patients in case we need it later.

**Comment by gingerich** [comment]
@clark-teeple Ya ideally we keep the flow as simple as possible for them. Initially this project started as a single-day hackathon so we decided to do without the patients table. I think if we add a unique index on patient phone number and [consistent formatting](https://www.npmjs.com/package/libphonenumber-js) then we can easily upsert patients on appointment creation without changing the flow. @iamstevedavis what do you think?

**Comment by iamstevedavis** [comment]
I don't know if a unique phone number would work because what if a mom and daughter are both in the system and the mom's cell phone is the contact for both, or a house phone for a family. I don't know if the use case for the app actually allows for that problem to arise (still a little hazy on the big picture). I wish we could get more data about a person like address and stuff because a combination of data might work. Name ( and I was thinking about splitting that out but I don't know if all cultures have both first and last names so I don't even know if that would work) and phone number as a composite might be an idea if we will run into the scenario above?

**Comment by iamstevedavis** [comment]
Based on my last comment I don't even know if having a patient table is the right approach. Maybe each appointment should be unique in and of itself and we keep the patient data in with the appointment. We can coalesce data based on phone number and patient name in conjunction with appointment data without having to worry about patient being unique kind of like they do when you call support. "We see a few Steve Davis' in here what is your address" and then they select the correct one based on some additional metadata. You have a better idea of the use case of the app though so what are your thoughts?

**Comment by iamstevedavis** [comment]
Really my question is what is the benefit of trying to force unique patients in the grand scheme of things? Is it actually necessary and worth the effort, especially for a proof of concept or alpha version.

**Comment by clark-teeple** [comment]
Keeping them in appointment would make it a lot simpler. @iamstevedavis and I have worked on projects with this requirement. It's a ton of work to do it right. It would be necessary if other organizations wanted to use the program, but it wouldn't be a problem introducing it later if need be as long as we don't do unique patients now.

**Comment by gingerich** [comment]
Good point, I would say there's no real advantage at this point to maintain unique patients. Appointments can be filtered by phone number (or phone + name combo) in order to recall appointments by patient. Based on the discussion I think it makes the most sense for now to keep patient information within the appointment itself.

**Comment by clark-teeple** [comment]
Should we close this PR then?

**Comment by KrisBraun** [comment]
I agree we can start without the concept of a patient. I do think we could introduce this when ready, modifying this table from patient to contact, and allowing contacts to have one or more patients.

**Comment by iamstevedavis** [comment]
I am going to close this based on the discussion above.

**Pull request closed without merging**